src/Console/ModelsCommand.php

```diff
@@ -502,47 +490,29 @@ protected function getTypeOverride($type)
      *
      * @param Model $model
      *
-     * @throws DBALException If custom field failed to register
      */
     public function getPropertiesFromTable($model)
     {
-        $database = $model->getConnection()->getDatabaseName();
-        $table = $model->getConnection()->getTablePrefix() . $model->getTable();
-        $schema = $model->getConnection()->getDoctrineSchemaManager();
-        $databasePlatform = $schema->getDatabasePlatform();
-        $databasePlatform->registerDoctrineTypeMapping('enum', 'string');
+        $table = $model->getTable();
+        $schema = $model->getConnection()->getSchemaBuilder();
+        $columns = $schema->getColumns($table);
+        $platformName = $model->getConnection()->getConfig('driver');
 
         if (strpos($table, '.')) {
             [$database, $table] = explode('.', $table);
         }
 
-        $platformName = $databasePlatform->getName();
-        $customTypes = $this->laravel['config']->get("ide-helper.custom_db_types.{$platformName}", []);
-        foreach ($customTypes as $yourTypeName => $doctrineTypeName) {
-            try {
-                if (!Type::hasType($yourTypeName)) {
-                    Type::addType($yourTypeName, get_class(Type::getType($doctrineTypeName)));
-                }
-            } catch (DBALException $exception) {
-                $this->error("Failed registering custom db type \"$yourTypeName\" as \"$doctrineTypeName\"");
-                throw $exception;
-            }
-            $databasePlatform->registerDoctrineTypeMapping($yourTypeName, $doctrineTypeName);
-        }
-
-        $columns = $schema->listTableColumns($table, $database);
-
         if (!$columns) {
             return;
         }
 
         $this->setForeignKeys($schema, $table);
         foreach ($columns as $column) {
-            $name = $column->getName();
+            $name = $column['name'];
             if (in_array($name, $model->getDates())) {
                 $type = $this->dateClass;
             } else {
-                $type = $column->getType()->getName();
+                $type = $column['type_name'];
```

Comment:
```suggestion
                $type = $this->laravel['config']->get("ide-helper.custom_db_types.{$driverName}.{$column['type_name']}", null);
                $type ??= $column['type'] === 'tinyint(1)' ? 'boolean' : null;
                $type ??= match ($column['type_name']) {
                    'varchar', 'nvarchar',
                    'char', 'bpchar', 'nchar',
                    'varbinary', 'blob', 'bytea',
                    'decimal', 'numeric',
                    'date', 'time', 'timetz', 'datetime', 'datetime2', 'datetimeoffset', 'timestamp', 'timestamptz', 'year',
                    'uuid', 'uniqueidentifier',
                    'json', 'jsonb',
                    'macaddr', 'inet',
                    'enum', 'set',
                    'geography', 'geometry', 'geometrycollection', 'linestring', 'multilinestring', 'multipoint', 'multipolygon', 'point', 'polygon',
                    'text', 'tinytext', 'longtext', 'mediumtext' => 'string',

                    'tinyint',
                    'integer', 'int', 'int4',
                    'smallint', 'int2',
                    'mediumint',
                    'bigint', 'int8' => 'integer',

                    'boolean', 'bool', 'bit' => 'boolean',

                    'float', 'real', 'float4',
                    'double', 'float8' => 'float',

                    default => 'mixed',
                };
```

Reply:
Almost all possible types for `mysql`, `mariadb`, `pgsql`, `sqlsrv`, and `sqlite`.

Reply:
Ah yes, but I think tinyint(1) is an integer for most drivers still? Only after casting it becomes a bool

Reply:
on MySQL, `bool` and `boolean` are synonyms for `tinyint(1)`, PHP also accepts `1` and `0` as boolean. But the mapping for all these edge cases (e.g. `year`, `binary`, `timestamp`, etc.) are a matter of personal preference IMO, it's up to you.

Reply:
Yeah but I mean, if you would `dump()` the attribute, it would show as integer, not as bool. Best practice would be to cast it to bool, but don't think Laravel does that for you.

Reply:
You're right, AFAIK Laravel doesn't cast it to bool.